operator/pkg/cli/generic.go

```diff
@@ -0,0 +1,109 @@
+package cli
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"os"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	yaml2 "k8s.io/apimachinery/pkg/util/yaml"
+)
+
+func createDecoder(data []byte) *yaml2.YAMLOrJSONDecoder {
+	var reader io.Reader
+	if len(data) > 0 {
+		reader = bytes.NewReader(data)
+	} else {
+		reader = io.Reader(os.Stdin)
+	}
+	dec := yaml2.NewYAMLOrJSONDecoder(reader, 10)
+	return dec
+}
+
+func getNextResource(dec *yaml2.YAMLOrJSONDecoder) (string, string, []byte, bool, error) {
```

Comment:
too many returns and it is hard to know what we are getting back. Consider adding at least description.